### PR TITLE
Add initial logger.

### PR DESCRIPTION
**Pull request by danpaz**
@sandinmyjoints Using the same strategy as pn-s3, still depends on `config`.
/cc @CarsonBanov

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,42 @@
...
+        level: 'info',
+        json: true,
+        prettyPrint: true
+      }
+    },
+    {
+      Loggly: {
+        level: 'warning',
+        json: true,
+        tag: 'pn-processing',
```
Comment (sandinmyjoints) [inline README.md]: Maybe put something all caps here like `'APP NAME'` to protect a bit against blind copying and pasting.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,42 @@
...
+## Usage
+
+```js
+var logging = require('pn-logging');
+logging.error(err.message, err);
+logging.info('Server started');
+
+// Or as middleware.
+app.use(logging.Log.middleware());
+```
```
Comment (sandinmyjoints) [inline README.md:51]: Another use case is an ad-hoc log using `new logging.Log()`.

**Comment by sandinmyjoints** [comment]
Nice! A few comments.

File: README.md (hunk 2 above)
Comment (danpaz) [inline README.md:51]: How/why would you use it in this way? For implementing custom levels?

**Comment by danpaz** [comment]
@sandinmyjoints Updated.

File: README.md (hunk 2 above)
Comment (sandinmyjoints) [inline README.md:51]: The logging middleware has a type of request by default. But sometimes you might want to log something out of band of the result. For example, in pn-processing, a request can contain multiple sqs records, each of which points to its own s3 item which will be fetched and handled independently. So you can give each of them their own `logging.Log` instance, set a custom type like `record`, add the s3 item key as meta -- all stuff that you couldn't do if you only emitted one log message for the entire request.

**Comment by sandinmyjoints** [comment]
LGTM. Approved.

**Pull request merged**